src/earn/prepareTransactions.ts

```diff
@@ -118,100 +123,169 @@ export async function prepareWithdrawAndClaimTransactions({
       ],
       ...shortcutTriggerArgs?.withdraw,
     })
+  Logger.debug(TAG, 'prepareWithdrawTransactions', {
+    withdrawTransactions,
+    pool,
+  })
+
+  // Is there a better way to do this?
+  // Pass a param to a shared prepareTransactions function to indicate that the transaction should also return swapTransaction:
+  return {
+    prepareTransactionsResult: await prepareTransactions({
+      feeCurrencies,
+      baseTransactions: rawShortcutTransactionsToTransactionRequests(withdrawTransactions),
+      isGasSubsidized: isGasSubsidizedForNetwork(networkId),
+      origin: 'earn-withdraw',
+    }),
+    swapTransaction: undefined,
+  }
+}
+
+// Used on the EarnConfirmationScreen.tsx
+export async function prepareWithdrawAndClaimTransactions({
```

Comment:
Is it possible for `prepareWithdrawAndClaimTransactions` to re-use both `prepareWithdrawTransactions` and `prepareClaimTransactions`?

Reply:
Something the code below works, but I get a return type error: `Property 'transactions' does not exist on type 'PreparedTransactionsResult'.` As this is a pre-existing function I think it's fine to leave as is.

```TypeScript
export async function prepareWithdrawAndClaimTransactions({
  walletAddress,
  feeCurrencies,
  pool,
  hooksApiUrl,
  rewardsPositions,
}: {
  walletAddress: Address
  feeCurrencies: TokenBalance[]
  pool: EarnPosition
  hooksApiUrl: string
  rewardsPositions: Position[]
}) {
  const { transactions: withdrawTransactions } = await prepareWithdrawTransactions({
    amount: pool.balance,
    walletAddress,
    feeCurrencies,
    pool,
    hooksApiUrl,
    useMax: true,
  })

  const { transactions: claimTransaction } = await prepareClaimTransactions({
    walletAddress,
    feeCurrencies,
    pool,
    hooksApiUrl,
    rewardsPositions,
  })

  Logger.debug(TAG, 'prepareWithdrawAndClaimTransactions', {
    withdrawTransactions,
    claimTransaction,
    pool,
  })

  const combinedTransactions = [...withdrawTransactions, ...claimTransaction]

  return prepareTransactions({
    feeCurrencies,
    baseTransactions: rawShortcutTransactionsToTransactionRequests(combinedTransactions),
    isGasSubsidized: isGasSubsidizedForNetwork(pool.networkId),
    origin: 'earn-claim-rewards',
  })
}
```